### PR TITLE
(#159) Shade `org.cactoos:cactoos` to jar

### PR DESCRIPTION
**Pull request by andreoss**
Per #159 
- Remove provided scope for cactoos
- Add shaded cactoos to jar

### CODE REVIEW

**Comment by victornoel** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/llorllale/cactoos-matchers/issues/196#issuecomment-693330617))

@victornoel This is what I know about this job in C63314D6Z, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#196](https://github.com/llorllale/cactoos-matchers/issues/196) is [in scope](http://www.zerocracy.com/policy.html#14) for 14hr
  * The role is `REV`
  * The job is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx) for [14hr](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @andreoss[/z](https://www.0crat.com/u/andreoss): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C63314D6Z?q=%7Bjob%3A%27gh%3Allorllale%2Fcactoos-matchers%23196%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

File: pom.xml [hunk 1]
```diff
@@ -153,6 +152,34 @@
...
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>3.2.4</version>
+        <executions>
+          <execution>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
```
Comment (fabriciofx) [inline pom.xml:165]: @andreoss WDYT add `<minimizeJar>true</minimizeJar>` configuration here? Check if the ubber-jar minimize its size. You can see more details [here](https://stackoverflow.com/questions/8817257/minimize-an-uber-jar-correctly-using-shade-plugin).

**Review by fabriciofx — CHANGES_REQUESTED** [review]
@andreoss Can you take a look, please?

**Comment by fabriciofx** [comment]
@andreoss it looks fine to me. @victornoel can you merge it, please?

**Comment by victornoel** [comment]
@andreoss there is a check failing, also I've you validated that it's working as expected, for example by:
- building the jar locally
- adding the dependency to cactoos
- running the tests

**Comment by fabriciofx** [comment]
@victornoel I'm seeing it now: linter phase was separated from build process. I didn't like it... :/

File: pom.xml (hunk 1 above)
Comment (andreoss) [reply pom.xml:165]: @fabriciofx Added

**Comment by andreoss** [comment]
@victornoel I don't see any log for this step. Qulice & xcop pass for me

**Comment by fabriciofx** [comment]
@andreoss Check the problem [here](https://github.com/llorllale/cactoos-matchers/pull/196/checks?check_run_id=1123689843).

**Comment by andreoss** [comment]
@fabriciofx Done

**Comment by fabriciofx** [comment]
@andreoss ok, thanks! @victornoel can you merge it, please?

**Comment by victornoel** [comment]
@andreoss have you validated that it's working as expected? For example by:
-  building the jar locally
-  adding the dependency to cactoos
-  running the tests

**Comment by andreoss** [comment]
@victornoel the tests pass fine with shaded jar from this branch.
We can have an integration for that

**Comment by victornoel** [comment]
@andreoss I'm not familiar with testing this kind of thing, do you have an idea how we could do that? If we add a todo it could be nice to give some insights.

**Comment by andreoss** [comment]
@victornoel I believe we can have a groovy script and run it with maven-invoker-plugin after install phase
https://github.com/jcabi/jcabi-parent/blob/master/pom.xml#L841
The script could either clone cactoos repository, compile it and run the tests, or grab jar with tests and run them without compilation.

**Comment by victornoel** [comment]
@andreoss I think let's forget about it for now :) if the need arises in the future, we will do it

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@fabriciofx please make sure you will find at least 3 issues during next CR, thanks!

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**